src/Chapter3.hs

```diff
@@ -374,6 +382,37 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight' = MkKnight
+   {knightHealth :: Int
+   ,knightAttack :: Int
+   ,knightGold :: Int
+   } deriving (Show)
+
+data Monster' = MkMonster
+   {monsterHealth :: Int
+   ,monsterAttack :: Int
+   ,monsterGold :: Int
+   } deriving (Show)
+
+-- fight :: Monster -> Knight -> Int
+-- We don't have to write the full fight only a strike
+-- That why the match can results in a draw ...
+-- fight m k = go (m, k) True
+--   where
+--     knightStrike  m k = (m {monsterHealth = knightAttack k}, k)
+--     monsterStrike m k = (m, k {knightHealth = monsterAttack m})
+--     stealMonster  m k = k {knightGold = knightGold k + monsterGold m}
+--     go (m, k) knightTurn
+--       | knightTurn  = if knightHealth k > 0 then go (knightStrike m k) False else -1
+--       | otherwise   = if monsterHealth m > 0
+--                       then go (monsterStrike m k) True
+--                       else knightGold (stealMonster m k)
+fight :: Monster' -> Knight' -> Int
+fight m k
+  | monsterHealth m <= knightAttack  k  = knightGold k + monsterGold m
+  | knightHealth  k <=  monsterAttack m = -1
+  | otherwise                           = knightGold k
```

Comment:
Your recursive implementation looks correct 👍🏻 
But yeah, this is the task at the beginning, so it is supposed to be simple 🙂